File: Python/TwoSum.py
def TwoSum(nums: list, target: int) -> list:
    sum_set = {}
    
    for i, num in enumerate(nums):
        if target - num in sum_set:
            return [i, sum_set[target-num]]
        sum_set[num] = i


if __name__ == '__main__':
    lista = [1, 2, 3, 4]
    print(TwoSum(lista, 7))

File: Java/TwoSum.java
import java.util.Map;
import java.util.HashMap;

public class TwoSum {
    public static void main(String[] args){

    }

    public static int[] twoSum(int[] nums, int target){
        Map<Integer, Integer> numToIndex = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            if(numToIndex.containsKey(target - nums[i]))
               return new int[] {numToIndex.get(target - nums[i]), i};
            numToIndex.put(nums[i], i);
        }

        throw new IllegalArgumentException();


    }
}